src/expression.rs

```diff
@@ -142,27 +139,37 @@ impl ops::Add<Term> for Expression {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Iter<'a> {
-    iterator: slice::Iter<'a, Term>,
+    slice: &'a [Term],
```

Comment:
`slice::Iter` does not support `PartialEq` or `Eq`.

Also, by storing the slice instead of the iterator, it is possible for internal crate functions to access the slice directly.

For example, in Earley parsing, it is able to store a reference to these `Term` slices directly, without any copying necessary.